components/ItemSummary.tsx

```diff
@@ -2,19 +2,25 @@
 import VoteButton from "@/islands/VoteButton.tsx";
 import type { Item } from "@/utils/db.ts";
 import UserPostedAt from "./UserPostedAt.tsx";
+import StaticVoteButton from "./StaticVoteButton.tsx";
 
 export interface ItemSummaryProps {
   item: Item;
   isVoted: boolean;
+  isSignedIn: boolean;
```

Comment:
Nice! I wonder if having `isVoted?: boolean` might be simpler, where it's `undefined` if the user is not signed in. WDYT?

Reply:
Yeah, I like that suggestion if reusing the `isVoted` prop. What do you think about `isVoted: boolean | undefined` instead since we probably want that prop to be required?

Reply:
`isVoted?: boolean` is the same as `isVoted: boolean | undefined` ([reference](https://www.typescriptlang.org/docs/handbook/2/functions.html#optional-parameters)). Having `isVoted?: boolean` would mean it is optional. Btw, if, for whatever reason, it's not simpler, let's stick to using `isSignedIn: boolean`.